File: services/shop-backend/src/main/java/io/miragon/shop/adapter/outbound/persistence/cart/CartEntity.java
package io.miragon.shop.adapter.outbound.persistence.cart;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "cart")
public class CartEntity {

    @Id
    @Column(name = "user_id", nullable = false)
    private String userId;

    @OneToMany(mappedBy = "cart", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    private List<CartItemEntity> items = new ArrayList<>();

    protected CartEntity() {}

    public CartEntity(String userId) {
        this.userId = userId;
    }

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public List<CartItemEntity> getItems() { return items; }
    public void setItems(List<CartItemEntity> items) { this.items = items; }
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/adapter/outbound/persistence/cart/CartEntity.kt
package io.miragon.shop.adapter.outbound.persistence.cart

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

@Entity(name = "cart")
class CartEntity(
    @Id
    @Column(name = "user_id", nullable = false)
    val userId: String,

    @OneToMany(mappedBy = "cart", cascade = [CascadeType.ALL], fetch = FetchType.LAZY, orphanRemoval = true)
    val items: MutableList<CartItemEntity> = mutableListOf()
) {
    companion object
}